src/compiler/diagnosticMessages.json

```diff
@@ -1859,10 +1859,22 @@
         "category": "Error",
         "code": 2553
     },
-    "Expected {0} type arguments, but got {1}.": {
+    "Expected at least {0} arguments, but got {1}.": {
         "category": "Error",
         "code": 2554
     },
+    "Expected {0} arguments, but got a minimum of {1}.": {
+        "category": "Error",
+        "code": 2555
+    },
+    "Expected at least {0} arguments, but got a minimum of {1}.": {
+        "category": "Error",
+        "code": 2556
+    },
+    "Expected {0} type arguments, but got {1}.": {

```

Comment:
"at least" indicates a rest argument, which TS doesn't have yet. I use a range, generated in the code, for optional [type] arguments, so you'd see a message like "Expected 1-3 type arguments, but got 4".